public/widgets/paper-badger-widget.js

```diff
@@ -87,22 +87,44 @@ function splitArray(arr, lookGroup, taxonomyClass)
 	return ttt;
 }
 
+function getEndpoint(confIn, count)
```

Comment:
Hi Rob,

Please don’t look at this from a semantic perspective.

Every function was chosen carefully and because of the function they provided.

“Push” in this case, ensures the correct order is maintained for every part of each possible endpoint supported.

“Join” collects and renders each part of the composition, without a need for loop and ensures they are separated by “/” – all this encapsulated in a single function, simple and elegant – keeping the code clean, free from confusing clutter.

And the last part acts as a test, so “/count” is only added as a suffix when required.

Kindest regards,



Al.

From: Robert Peters [mailto:notifications@github.com]
Sent: 24 September 2015 19:42
To: mozillascience/PaperBadger
Cc: Coleman, Alistair
Subject: Re: [PaperBadger] Added support for various endpoints to Paper Badget widget. (#131)


In public/widgets/paper-badger-widget.js<https://github.com/mozillascience/PaperBadger/pull/131#discussion_r40356064>:

> @@ -87,22 +87,44 @@ function splitArray(arr, lookGroup, taxonomyClass)

>       return ttt;

>  }

>

> +function getEndpoint(confIn, count)

Code wise this works. Style wise I find having both + and join('/') confusing:

    function getEndpoint(confIn, count) {

        var endPoint=[];

        endPoint.push("https://badges.mozillascience.org");

        if (confIn["article-doi"]) endPoint.push("papers/" + confIn["article-doi"]);

        if (confIn["orcid"]) endPoint.push("users/" + confIn["orcid"]);

        endPoint.push("/badges");

        if (count) endPoint.push("/count" + count)

        return endPoint.join("/");

    }



    function getEndpoint(confIn, count) {

        var endPoint = "https://badges.mozillascience.org";

        if (confIn["article-doi"]) endPoint += "papers/" + confIn["article-doi"];

        if (confIn["orcid"]) endPoint += "users/" + confIn["orcid"];

        endPoint += "/badges";

        if (count) endPoint += "/count" + count;

        return endPoint;

    }

Would either of those be clearer? I'm not sure.

—
Reply to this email directly or view it on GitHub<https://github.com/mozillascience/PaperBadger/pull/131/files#r40356064>.